tests/python_functional/src/syslog_ng_config/syslog_ng_config.py

```diff
@@ -65,7 +65,10 @@ def write_config(self, config_path):
         else:
             rendered_config = ConfigRenderer(self.__syslog_ng_config).get_rendered_config()
         logger.info("Generated syslog-ng config\n{}\n".format(rendered_config))
-        FileIO(config_path).rewrite(rendered_config)
+        f = File(config_path)
```

Comment:
You are absolutely right, we can use the create_file() here instead of write this 4 lines (create + write + close file).
There is one minor problem, in create_file() there is a check if the written file is already exist or not, and the function raises if the file exist. But we in syslog_ng_config.py file need to overwrite the original file (before syslog-ng reload, config could changed), so if we start using create_file() here, we should extend it with a new parameter: 'overwrite=False|True', or we should create a new function which will overwrite the existing file: overwrite_file(file, content). 
What do you think, which solution would be better?

Reply:
IMHO neither of them. My idea was bad.
As far as I understand `common.operations` is functioning as a wrapper in case some python2/python3 compatibility and as a module for high-level helper functions.

I would remove `create_file` from `common.operations` as nobody uses it currently, and also because I don't see any reason why `create_file` couldn't be put into the `File` class.
Nevertheless, don't put `create_file` into `File` either, please. 

I would fancy if `File` could be used within a `with` statement:
```
with File(config_path, "w+") as config_file:
    config_file.write(rendered_config)
```
This is optional to do in this PR, I'm okay with the current form, we can enhance the `File` class when we see the need for it, but please remove `create_file` from `common.operations`

Reply:
I am ok with removing `create_file()` from `common.operations` and replace it `with` statement (as you suggested),  but if there will be another place where we should create a file with some content I would put this method to the `File()` class (but would we remember for this?).
What do you think?

Reply:
Yes, we should remove `create_file` from `common.operations`, no one uses it.
Thanks if you like the idea of the `with` statement. I would totally accept if you would like to do it in another PR.

About the question of future use cases of `create_file` method: I don't think it's needed, especially after having the `with` statement support, but that's just my opinion (although this PR doesn't use `create_file` in any place either). 
Users can see the interface of the `File` class (e.g. the `open`, `close` methods) so we would remember it.

Reply:
I have fixed this.